### ISSUE TITLE
Documents with MBC are not handled properly in lsp

### ISSUE THREAD

**Issue by kitsonk** [denoland/deno#8753]
Editing, or creating a document with the following text causes all sorts of issues and lsp server crashes:

```ts
const foo = `编写软件很难`;

console.log(foo);
```

For example, changing `.log` to `.warn` will result in bad diagnostics like:

```
Unterminated template literal.
```

This is because characters are not being counted the same between what is being sent by the lsp and how the lsp internally calculates them, causing the internal file to be updated in the wrong location.

**Comment by uki00a** [comment]
I ran into the same problem. I'll try to fix this.

**Comment by kitsonk** [comment]
Fixed in #9151
